File: packages/jsii-pacmak/test/expected.dotnet.jsii-calc/AWS.Jsii.Tests.Calculator/Statics.cs
using AWS.Jsii.Runtime.Deputy;
using System.Collections.Generic;

namespace AWS.Jsii.Tests.Calculator
{
    [JsiiClass(typeof(Statics), "jsii-calc.Statics", "[{\"name\":\"value\",\"type\":{\"primitive\":\"string\"}}]")]
    public class Statics : DeputyBase
    {
        public Statics(string value): base(new DeputyProps(new object[]{value}))
        {
        }

        protected Statics(ByRefValue reference): base(reference)
        {
        }

        protected Statics(DeputyProps props): base(props)
        {
        }

        [JsiiProperty("value", "{\"primitive\":\"string\"}")]
        public virtual string Value
        {
            get => GetProperty<string>();
        }

        /// <summary>Jsdocs for static property.</summary>
        [JsiiProperty("Foo", "{\"primitive\":\"string\"}")]
        public virtual string Foo
        {
            get => GetProperty<string>();
        }

        /// <summary>Constants may also use all-caps.</summary>
        [JsiiProperty("BAR", "{\"primitive\":\"number\"}")]
        public virtual double BAR
        {
            get => GetProperty<double>();
        }

        /// <summary>Constants can also use camelCase.</summary>
        [JsiiProperty("zooBar", "{\"collection\":{\"kind\":\"map\",\"elementtype\":{\"primitive\":\"string\"}}}")]
        public virtual IDictionary<string, string> ZooBar
        {
            get => GetProperty<IDictionary<string, string>>();
        }

        /// <summary>
        /// Jsdocs for static getter.
        /// Jsdocs for static setter.
        /// </summary>
        [JsiiProperty("instance", "{\"fqn\":\"jsii-calc.Statics\"}")]
        public virtual Statics Instance
        {
            get => GetProperty<Statics>();
            set => SetProperty(value);
        }

        [JsiiProperty("nonConstStatic", "{\"primitive\":\"number\"}")]
        public virtual double NonConstStatic
        {
            get => GetProperty<double>();
            set => SetProperty(value);
        }

        [JsiiProperty("ConstObj", "{\"fqn\":\"jsii-calc.DoubleTrouble\"}")]
        public virtual DoubleTrouble ConstObj
        {
            get => GetProperty<DoubleTrouble>();
        }

        /// <summary>Jsdocs for static method</summary>
        /// <param name = "name">The name of the person to say hello to</param>
        [JsiiMethod("staticMethod", "{\"primitive\":\"string\"}", "[{\"name\":\"name\",\"type\":{\"primitive\":\"string\"}}]")]
        public virtual string StaticMethod(string name)
        {
            return InvokeMethod<string>(new object[]{name});
        }

        [JsiiMethod("justMethod", "{\"primitive\":\"string\"}", "[]")]
        public virtual string JustMethod()
        {
            return InvokeMethod<string>(new object[]{});
        }
    }
}

File: packages/jsii-pacmak/test/expected.jsii-calc/java/src/main/java/software/amazon/jsii/tests/calculator/compliance/Statics.java
package software.amazon.jsii.tests.calculator.compliance;

/**
 * EXPERIMENTAL
 */
@javax.annotation.Generated(value = "jsii-pacmak")
@software.amazon.jsii.Stability(software.amazon.jsii.Stability.Level.Experimental)
@software.amazon.jsii.Jsii(module = software.amazon.jsii.tests.calculator.$Module.class, fqn = "jsii-calc.compliance.Statics")
public class Statics extends software.amazon.jsii.JsiiObject {

    protected Statics(final software.amazon.jsii.JsiiObjectRef objRef) {
        super(objRef);
    }

    protected Statics(final software.amazon.jsii.JsiiObject.InitializationMode initializationMode) {
        super(initializationMode);
    }

    static {
        BAR = software.amazon.jsii.JsiiObject.jsiiStaticGet(software.amazon.jsii.tests.calculator.compliance.Statics.class, "BAR", java.lang.Number.class);
        CONST_OBJ = software.amazon.jsii.JsiiObject.jsiiStaticGet(software.amazon.jsii.tests.calculator.compliance.Statics.class, "ConstObj", software.amazon.jsii.tests.calculator.compliance.DoubleTrouble.class);
        FOO = software.amazon.jsii.JsiiObject.jsiiStaticGet(software.amazon.jsii.tests.calculator.compliance.Statics.class, "Foo", java.lang.String.class);
        ZOO_BAR = java.util.Collections.unmodifiableMap(software.amazon.jsii.JsiiObject.jsiiStaticGet(software.amazon.jsii.tests.calculator.compliance.Statics.class, "zooBar", software.amazon.jsii.NativeType.mapOf(software.amazon.jsii.NativeType.forClass(java.lang.String.class))));
    }

    /**
     * EXPERIMENTAL
     * <p>
     * @param value This parameter is required.
     */
    @software.amazon.jsii.Stability(software.amazon.jsii.Stability.Level.Experimental)
    public Statics(final @org.jetbrains.annotations.NotNull java.lang.String value) {
        super(software.amazon.jsii.JsiiObject.InitializationMode.JSII);
        software.amazon.jsii.JsiiEngine.getInstance().createNewObject(this, new Object[] { java.util.Objects.requireNonNull(value, "value is required") });
    }

    /**
     * Jsdocs for static method.
     * <p>
     * EXPERIMENTAL
     * <p>
     * @param name The name of the person to say hello to. This parameter is required.
     */
    @software.amazon.jsii.Stability(software.amazon.jsii.Stability.Level.Experimental)
    public static @org.jetbrains.annotations.NotNull java.lang.String staticMethod(final @org.jetbrains.annotations.NotNull java.lang.String name) {
        return software.amazon.jsii.JsiiObject.jsiiStaticCall(software.amazon.jsii.tests.calculator.compliance.Statics.class, "staticMethod", java.lang.String.class, new Object[] { java.util.Objects.requireNonNull(name, "name is required") });
    }

    /**
     * EXPERIMENTAL
     */
    @software.amazon.jsii.Stability(software.amazon.jsii.Stability.Level.Experimental)
    public @org.jetbrains.annotations.NotNull java.lang.String justMethod() {
        return this.jsiiCall("justMethod", java.lang.String.class);
    }

    /**
     * Constants may also use all-caps.
     * <p>
     * EXPERIMENTAL
     */
    @software.amazon.jsii.Stability(software.amazon.jsii.Stability.Level.Experimental)
    public final static java.lang.Number BAR;

    /**
     * EXPERIMENTAL
     */
    @software.amazon.jsii.Stability(software.amazon.jsii.Stability.Level.Experimental)
    public final static software.amazon.jsii.tests.calculator.compliance.DoubleTrouble CONST_OBJ;

    /**
     * Jsdocs for static property.
     * <p>
     * EXPERIMENTAL
     */
    @software.amazon.jsii.Stability(software.amazon.jsii.Stability.Level.Experimental)
    public final static java.lang.String FOO;

    /**
     * Constants can also use camelCase.
     * <p>
     * EXPERIMENTAL
     */
    @software.amazon.jsii.Stability(software.amazon.jsii.Stability.Level.Experimental)
    public final static java.util.Map<java.lang.String, java.lang.String> ZOO_BAR;

    /**
     * Jsdocs for static getter.
     * <p>
     * Jsdocs for static setter.
     * <p>
     * EXPERIMENTAL
     */
    @software.amazon.jsii.Stability(software.amazon.jsii.Stability.Level.Experimental)
    public static @org.jetbrains.annotations.NotNull software.amazon.jsii.tests.calculator.compliance.Statics getInstance() {
        return software.amazon.jsii.JsiiObject.jsiiStaticGet(software.amazon.jsii.tests.calculator.compliance.Statics.class, "instance", software.amazon.jsii.tests.calculator.compliance.Statics.class);
    }

    /**
     * Jsdocs for static getter.
     * <p>
     * Jsdocs for static setter.
     * <p>
     * EXPERIMENTAL
     */
    @software.amazon.jsii.Stability(software.amazon.jsii.Stability.Level.Experimental)
    public static void setInstance(final @org.jetbrains.annotations.NotNull software.amazon.jsii.tests.calculator.compliance.Statics value) {
        software.amazon.jsii.JsiiObject.jsiiStaticSet(software.amazon.jsii.tests.calculator.compliance.Statics.class, "instance", java.util.Objects.requireNonNull(value, "instance is required"));
    }

    /**
     * EXPERIMENTAL
     */
    @software.amazon.jsii.Stability(software.amazon.jsii.Stability.Level.Experimental)
    public static @org.jetbrains.annotations.NotNull java.lang.Number getNonConstStatic() {
        return software.amazon.jsii.JsiiObject.jsiiStaticGet(software.amazon.jsii.tests.calculator.compliance.Statics.class, "nonConstStatic", java.lang.Number.class);
    }

    /**
     * EXPERIMENTAL
     */
    @software.amazon.jsii.Stability(software.amazon.jsii.Stability.Level.Experimental)
    public static void setNonConstStatic(final @org.jetbrains.annotations.NotNull java.lang.Number value) {
        software.amazon.jsii.JsiiObject.jsiiStaticSet(software.amazon.jsii.tests.calculator.compliance.Statics.class, "nonConstStatic", java.util.Objects.requireNonNull(value, "nonConstStatic is required"));
    }

    /**
     * EXPERIMENTAL
     */
    @software.amazon.jsii.Stability(software.amazon.jsii.Stability.Level.Experimental)
    public @org.jetbrains.annotations.NotNull java.lang.String getValue() {
        return this.jsiiGet("value", java.lang.String.class);
    }
}
